pages/[league]/players.tsx

```diff
@@ -41,28 +44,56 @@ function PlayerPage({ league }: Props): JSX.Element {
     isLoading: isLoadingGoalies,
   } = useGoalieRatings(league);
 
+  // search functions
+  const searchTypes: Array<string> = ["Name", "Position"]
+
+  const [searchText, setSearchText] = useState('')
+  const [searchType, setSearchType] = useState(searchTypes[0])
+
+  const updateSearchText = (text) => {
+    setSearchText(text)
+  }
+
+  const updateSearchType = (searchType) => {
+    setSearchType(searchType)
+  }
+
+  const { players: filteredPlayerRatings, goalies: filteredGoalieRatings } = searchBarFilterPlayers({
+    searchText: searchText, 
+    searchType: searchType, 
+    players: skaterratings, 
+    goalies: goalieratingdata
+    })
+
+  const { players: filteredSkaters, goalies: filteredGoalies } = searchBarFilterPlayers({
+    searchText: searchText, 
+    searchType: searchType, 
+    players: skater, 
+    goalies: goalie
+    })
+
   const getSkaters = () =>
-    skaterratings
```

Comment:
replace the existing objects with the filtered versions.  If there is nothing the search for, the filtered objects match the original objects. i.e. no filtering

Reply:
React Table includes a `useFilter` hook which may make this a bit more optimized. The problem with this is that it needs to re-render the whole table every time something is typed, so we may want to try to go with their solution instead for the rendering of the data. 

https://react-table.tanstack.com/docs/api/useFilters

Reply:
okay, I will change it to use that hook.  This will take some time as it is kind of a complete rework.  Not a big deal, this is how I learn.